src/JSInterop/Microsoft.JSInterop.JS/src/src/Microsoft.JSInterop.ts

```diff
@@ -430,7 +451,22 @@ export module DotNet {
     }
   }
 
-  function argReplacer(key: string, value: any) {
-    return value instanceof DotNetObject ? value.serializeAsArg() : value;
+  function stringifyArgs(args: any[] | null) {
+    let byteArrayIndex = 0;
+
+    return JSON.stringify(args, argReplacer);
+
+    function argReplacer(key: string, value: any) {
+      if (value instanceof DotNetObject) {
+        return value.serializeAsArg();
+      } else if (value instanceof Uint8Array) {
+        dotNetDispatcher!.supplyByteArray(byteArrayIndex, value);
+        const jsonValue = { [byteArrayRefKey]: byteArrayIndex };
+        byteArrayIndex++;
+        return jsonValue;
+      }
+
+      return value;
+    }
```

Comment:
TBH I find this refactoring slightly unexpected. Would it also work to track `byteArrayIndex` outside the function as a file-level variable (just like `nextJsObjectId`), and avoid the extra layer of function wrapping around `argReplacer`?

Reply:
Ah no sorry, I realise why you did this now (to reset the ID to `0` at the start of each serialization). This makes sense!

Reply:
I guess it would still be a tiny bit more conventional not to create a new `argReplacer` closure instance on each call, by doing something like:

```ts
let nextbyteArrayIndex: number;

function stringifyArgsOrReturnValue(args: any[] | null) {
    nextbyteArrayIndex = 0;
    return JSON.stringify(args, argReplacer);
}

function argReplacer(...) { ... unchanged ... }
```

[This doc](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Closures#performance_considerations) seems to back this up, but not really a big deal AFAIK!

Reply:
> not to create a new argReplacer closure instance on each call

Ahh great catch, thanks!